### ISSUE TITLE
Getting Rickshawgraph working in a custom widget/dashboard.

### ISSUE THREAD

**Issue by wolfspyre** [Shopify/dashing#527]
I’m a little confused as to where else to direct this question. Suggestions are welcome in that direction as well.

I’m trying to use [jwalton’s Rickshawgraph widget](https://gist.github.com/jwalton/6614023) working in a custom widget/dashboard I’m making for monitoring the state of a 3d printer controlled by [octoprint](http://octoprint.org).

[Here’s a gist](https://gist.github.com/wolfspyre/b72a129d6b616f0311cf) of the work in progress dashboard, and relevant job I’ve created, along with some logged debug data before triggering send_event

Based on what’s mentioned in the [passing data section of the Rickshawgraph Readme.md](https://gist.github.com/jwalton/6614023#passing-data) I think I’m constructing the data correctly, but obviously I’m not.

I’ve looked around for functional examples but couldn’t quite piece together what I’m doing wrong here.

I’m sure it’s something utterly pedestrian that I’m overlooking.

 Does anyone mind pointing me in the right direction?

Thanks a bunch!

**Comment by jwalton** [comment]
Hi @wolfspyre.  Sorry for not seeing your comments on my gist; for some reason I never get notifications when someone comments on a gist.

Ok, here's what I did:

* Installed the latest dashing with `gem install dashing`.
* Created new dashboard with `dashing new db`.
* Copied your dashboard.erb file, created a dummy job that sends the logged data from your gist:

        SCHEDULER.every '2s' do
            bed_graphite = [{:target=>"octoprint_bed.temp.actual", :datapoints=>[[9, 1425294886], ...]}]
            send_event('octoprint_bed_graph', series: bed_graphite)
        end

* Copied the rickshawgraph files into db/widgets/rickshawgraph/*.

When I tried this, I ran into a few problems:

1) There was a bug in rickshawgraph which was obscuring the real problem here, so I fixed that (along with a couple of other comments people made in my gist.)
2) The real problem, though, is that you're setting `data-renderer="multigraph"`, but this isn't a [valid Rickshaw renderer](https://github.com/shutterstock/rickshaw#renderer).  Try setting `data-renderer="line"` and you'll get graphs like this:

![screen shot 2015-03-03 at 3 24 27 pm](https://cloud.githubusercontent.com/assets/1771003/6471498/6f251126-c1b9-11e4-83c8-a2c02ba57560.png)

**Comment by wolfspyre** [comment]
Hi there,

Thanks for your time!

Crap! I'd forgotten to fix that in my gist. Sorry for the red herring!!!!

Even with the data-renderer set to line, I still have completely blank
graphs:
https://www.evernote.com/shard/s255/sh/75907b01-8e2b-494a-ad22-d5109a6be9f2/b37349c91821fd1fb5873ef68f871196

I don't think that's the problem.

**Comment by wolfspyre** [comment]
After a little offline back-and-forth, I've successfully gotten this
working.

The problem was that in attempting to integrate the Rickshawgraph plugin, I
must have broken something, so it wasn't properly generating graphs.

I pulled down the exact Rickshawgraph coffee/html/scss files from
https://gist.github.com/jwalton/6614023 and placed them in
widgets/rickshawgraph; restarted my  dashboard, and viola
https://www.evernote.com/shard/s255/sh/25fcc9c9-7f73-4206-bc9b-4842a0fbeb78/d163a14714e060a90f3f7e21103986c1

Functional Graphs!!!!

Closing this issue. I'll post another comment once I release the fully
implemented/functional dashboard/widget/job collection so that others might
not run into the same difficulty I did.
